helpers/nodeHelpers.js

```diff
@@ -0,0 +1,18 @@
+
+function checkValidWriteInput(voltage, node) {
+    const upperLim = 10
+    const lowerLim = 0
+    if(typeof(voltage) === "number" && (voltage <= upperLim && voltage >= lowerLim) ){
+        return true 
+    }
+    else{
+        node.status({fill:"red", shape:"ring", text:"please enter a valid number to write voltage"});
+    }
+
+    return false;
+      
+}
```

Comment:
on the second thought, the upper limit changes when the gain changes. may  be Upper/Lower limits should be parameters passed.

Reply:
"does thow new Error('Failed to initialize dac.') on line 25 terminates the program? if not, what happens if the program reaches to line 31 after the error is thrown"  --> No I tested this and it does not terminate the program. It simply makes the node unusable. The node is still be visible with the error status and debug console will show the error message but node-red continues to work fine.

Reply:
> on the second thought, the upper limit changes when the gain changes. may be Upper/Lower limits should be parameters passed.

cd1db0f78434f35c83c1c3746d25b2556acfaed4